packages/rspack/src/config/zod.ts

```diff
@@ -804,7 +804,7 @@ const externalItemValue = z
 			root: z.string().or(z.string().array()),
 			commonjs: z.string().or(z.string().array()),
 			commonjs2: z.string().or(z.string().array()),
-			amd: z.string().or(z.string().array()).optional()
+			amd: z.string().or(z.string().array())
```

Comment:
I prefer to validate `amd` only when the externalType is `umd`, because other external types do not need an `amd` property.